PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/AzureCoreToolsMissingNupkgInstaller.kt

```diff
@@ -140,19 +141,21 @@ class AzureCoreToolsMissingNupkgInstaller : StartupActivity {
     }
 }
 
-fun ProjectModelViewHost.findProjectsWithVirtualFile(virtualFile: VirtualFile): List<ProjectModelNode> {
-    val visitor = object : ProjectModelNodeVisitor() {
-        val projects: HashSet<ProjectModelNode> = hashSetOf()
+fun WorkspaceModel.findProjectsWithVirtualFile(virtualFile: VirtualFile): List<ProjectModelEntity> {
+    val visitor = object : ProjectModelEntityVisitor() {
+        val projects: HashSet<ProjectModelEntity> = hashSetOf()
 
-        override fun visitNode(node: ProjectModelNode): Result {
-            if (node.getVirtualFile() == virtualFile) {
-                node.containingProject()?.let { projects.add(it) }
+        override fun visitNode(entity: ProjectModelEntity): Result {
+            if (entity.url?.virtualFile == virtualFile) {
```

Comment:
I did not test this, but I believe it is `getVirtualFileAsContentRoot()` API for this case.

Reply:
It seems we are looking for any project with specific files inside. I suggest write something like that: 
```
WorkspaceModel.getInstance().getProjectModelEntities(virtualFile, project).map { it.containingProjectEntity() }
```

Reply:
The file is not yet in the model at this point :(

Reply:
is there a way to "refresh" the model?

Reply:
Nevermind, found a way to make this all work. Updates incoming!